### ISSUE TITLE
Observable.from -->  object is not observable

### ISSUE THREAD

**Issue by ocombe** [ReactiveX/RxJS#1003]
Is there a reason why we cannot create an Observable from an object?
It was possible before using Observable.of, but now with from it doesn't seem possible anymore ?

**Comment by kwonoj** [comment]
Would you able to share small snippet to describe issues you're experiencing?

**Comment by ocombe** [comment]
On second thought, this might be a problem with me using RxJS alpha 11 (because of Angular 2 currently locked down to this version) and being unable to use Observable.of
Are .of & .from different ?

**Comment by kwonoj** [comment]
> Are .of & .from different ? 

: sort of. 

`Observable.from = FromObservable.create;`

and 

`Observable.of = ArrayObservable.of;`

currently. (Latest tot of RxJS5 at least).

**Comment by ocombe** [comment]
Ok, so there's no way to use ArrayObservable.of on an Object (that is not an array) then ?

**Comment by kwonoj** [comment]
I still would like to ask usecases you're trying, in typical usecases I didn't see it's blocking my usage for instance 

`Observable.of(1).subscribe(console.log)` 

would work. Curious what object you're trying to pass.

**Comment by kwonoj** [comment]
> Curious what object you're trying to pass.

: reason that it might have some of bugs couldn't be caught until. :)

**Comment by ocombe** [comment]
Sure, the code (just updated) is here: https://github.com/ocombe/ng2-translate/blob/master/src/translate.service.ts#L112

I'm returning an observable that can something be pending (because of an http request), and sometimes resolve immediately once the request has been made at least once (or when it's not pending).
The observable can be made of a promise, or the actual object return by the http request.

**Comment by blesh** [comment]
@ocombe 

`from` accepts any Iterable (Arrays, Strings, Generators, Custom Iteratables, etc), Promise, Observable, or "lowercase o" observable.

`of` takes a bunch of values and makes them into an Observable of those values.

If you want to observe an arbitrary object, that object needs to implement `Symbol.observable` and return a (lowercase "o") observable. That is an object with a subscribe method that accepts an observer.

**Comment by ocombe** [comment]
I don't want to observe the object, I just need to return an observable what ever the status of my request is.
It's well explained in http://reactivex.io/documentation/operators/from.html :
> When you work with Observables, it can be more convenient if all of the data you mean to work with can be represented as Observables, rather than as a mixture of Observables and other types. This allows you to use a single set of operators to govern the entire lifespan of the data stream.

**Comment by blesh** [comment]
@ocombe it's a little hard to follow the code you sent because it's very heavily abstracted. Perhaps it would be valuable to reduce the abstraction until you get it working, then factor your code out as necessary for testing? FWIW: Angular2's http.request() should just return an `Observable<Response>`

**Comment by blesh** [comment]
`Observable.of` will be a completely synchronous observable of whatever value you give it.

`Observable.from` converts from some type (Promise, Array, Iterable, Observable, lowercase-o observable) to an Rx.Observable.

**Comment by ocombe** [comment]
Yeah that's why I used the ArrayObservable.of in the end, and it works
fine, it's just weird to have use something called ArrayObservable to make
an Observable from any static value that is not always an array.

**Comment by kwonoj** [comment]
> something called ArrayObservable to make an Observable from any static value that is not always an array.

: Any reason for not using `Observable.of`? it's for generic interface for values.

**Comment by ocombe** [comment]
It's not available on alpha 11 and Angular 2 is locked down on this version
for now, I'll switch to that once they update their dependencies :-)

**Comment by ocombe** [comment]
Thanks for the explanations anyway, I'll close this issue since it's not really an issue.
By the way, it would be really nice to have documentation for this project, the current docs in the repo are really really light when they are available at all (all the links on https://github.com/ReactiveX/RxJS/blob/master/doc/index.md are 404)

**Comment by kwonoj** [comment]
For document, totally agree. That's one of milestone should be achieved we're trying to push for.

**Comment by ocombe** [comment]
:+1:

**Comment by aravindfz** [comment]
@ocombe what is the equivalent for the `Observable.of` in the recent version of rxjs. I'm using `version 5.5.5`

**Comment by ocombe** [comment]
`import {of} from "rxjs/observable/of";`

https://github.com/ngx-translate/core/blob/master/lib/src/translate.service.ts#L11

**Comment by benlesh** [comment]
@aravindfz `Observable.of` will still work. We didn't remove `rxjs/add/observable/of`... not even in the v6 alpha.

But in ALL versions of 5.x you can import from `import { of } from 'rxjs/observable/of'`, it's nothing new to 5.5, it's just that word didn't get out that this is probably better until then.
